lib/cppcheck.cpp

```diff
@@ -763,11 +769,15 @@ void CppCheck::reportErr(const ErrorLogger::ErrorMessage &msg)
     const Suppressions::ErrorMessage errorMessage = msg.toSuppressionsErrorMessage();
 
     if (mUseGlobalSuppressions) {
-        if (mSettings.nomsg.isSuppressed(errorMessage))
+        if (mSettings.nomsg.isSuppressed(errorMessage)) {
+            mSuppressExitCode = true;
```

Comment:
Does this mean that if some message is suppressed then the exit code will be 0?

Reply:
yeh, i follow expect in issue, if suppression option on, suppression's error will be ignored, so exit code will be 0

Reply:
Maybe I misunderstand this..

Imagine there are 1000 errors. And you suppress 1 of those. So the exit code will be 0 even though there are 999 unsuppressed errors?

Reply:
I think I misunderstood.. All old tests pass.

Reply:
Example code (I put this code in 1.cpp):

    void f(int x, int y) {
        a = x / 0;
        b = y / 0;
    }

There are two divisions by zero, at line 2 and line 3.

Suppressing the last warning means that the exit code is 0:

    $ ./cppcheck --error-exitcode=1 --suppress=zerodiv:1.cpp:3 1.cpp
    Checking 1.cpp ...
    [1.cpp:2]: (error) Division by zero.

    $ echo $?
    0